testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/DeploymentInfoTestCase.java

```diff
@@ -0,0 +1,67 @@
+package org.jboss.as.test.integration.management.cli;
+
+import org.jboss.as.cli.CommandContext;
+import org.jboss.as.cli.Util;
+import org.jboss.as.test.integration.management.base.AbstractCliTestBase;
+import org.jboss.as.test.integration.management.util.CLITestUtil;
+import org.jboss.dmr.ModelNode;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.core.testrunner.WildflyTestRunner;
+
+/**
+ *
+ * @author baranowb
+ *
+ */
+@RunWith(WildflyTestRunner.class)
+public class DeploymentInfoTestCase extends AbstractCliTestBase {
+
+    private CommandContext ctx;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        AbstractCliTestBase.initCLI();
+    }
+
+    @AfterClass
+    public static void afterClass() throws Exception {
+        AbstractCliTestBase.closeCLI();
+    }
+
+    @Before
+    public void before() throws Exception {
+        ctx = CLITestUtil.getCommandContext();
+        ctx.connectController();
+    }
+
+    @After
+    public void after() throws Exception {
+        ctx.terminateSession();
+    }
+
+    @Test
+    public void testBasicPathArgumentRecognitionAndParsing() throws Exception {
+        ModelNode request = ctx.buildRequest("deployment-info --headers");
+        Assert.assertFalse(request.hasDefined(Util.OPERATION_HEADERS));
+    }
+
+    @Test
+    public void testBasicPathArgumentRecognitionAndParsing2() throws Exception {
+        ModelNode request = ctx.buildRequest("deployment-info --headers=false");
+        Assert.assertFalse(request.hasDefined(Util.OPERATION_HEADERS));
```

Comment:
Actually this should result in a CLI error after all since --headers, --headers= and --headers=<any_simple_value> don't make sense according to https://docs.jboss.org/author/display/WFLY10/Admin+Guide#AdminGuide-OperationHeaders
So when we check the value of headers in BaseOperationCommand.addHeaders, unless it's an OBJECT type we should not accept it.